src/v0/sources/ortto/transform.js

```diff
@@ -0,0 +1,114 @@
+const path = require('path');
+const fs = require('fs');
+const { flattenJson, removeUndefinedAndNullValues, generateUUID } = require('../../util');
+const Message = require('../message');
+const eventMapping = require('./event_mapping.json');
+const { JSON_MIME_TYPE } = require('../../util/constant');
+
+// import mapping json using JSON.parse to preserve object key order
+const mapping = JSON.parse(fs.readFileSync(path.resolve(__dirname, './mapping.json'), 'utf-8'));
+
+function settingProperties(event, message) {
+  const messageReplica = message;
+
+  // flattening the event and assigning it to properties
+  messageReplica.properties = removeUndefinedAndNullValues(flattenJson(event));
+
+  // fields that are already mapped
+  const excludeFields = [
+    'contact.email',
+    'contact.contact_id',
+    'id',
+    'time',
+    'activity.field_id',
+    'contact.external_id',
+    'contact.country.name',
+    'contact.city.name',
+    'contact.postal',
+    'contact.first_name',
+    'contact.last_name',
+    'contact.birthday.day',
+    'contact.birthday.month',
+    'contact.birthday.year',
+    'contact.phone_number.c',
+    'contact.phone_number',
+    'contact.phone_number.n',
+  ];
+
+  // deleting already mapped fields
+  excludeFields.forEach((field) => {
+    delete messageReplica.properties[field];
+  });
+
+  return message;
+}
+
+function process(event) {
+  let message = new Message(`ortto`);
+
+  // Here, we are checking for the test event to discard them
+  if (event.activity?.field_id === 'act::test_webhook') {
+    return {
+      outputToSource: {
+        body: Buffer.from(JSON.stringify(event)).toString('base64'),
+        contentType: JSON_MIME_TYPE,
+      },
+      statusCode: 200,
+    };
+  }
+
+  // we are setting event type as track always
+  message.setEventType('track');
+
+  message.setPropertiesV2(event, mapping);
+
+  // setting anonymousId
+  if (!message.userId) {
+    message.anonymousId = generateUUID();
+  }
+
+  if (event.contact?.birthday) {
+    const month =
+      event.contact.birthday?.month < 10
```

Comment:
How are we planning to handle if `event.contact.birthday` is a string ?

Reply:
This is the Ortto source format, should we expect any mismatch in that?

Reply:
Just checking with you if the `contact.birthday` can be a string.